fe/fe-core/src/main/java/com/starrocks/analysis/ListPartitionDesc.java

```diff
@@ -0,0 +1,192 @@
+                }
+            }
+        }
+    }
+
+    public List<String> findAllParitionName() {
+        List<String> partitionNames = new ArrayList<>();
+        this.singleListPartitionDescs.forEach(desc -> partitionNames.add(desc.getPartitionName()));
+        this.multiListPartitionDescs.forEach(desc -> partitionNames.add(desc.getPartitionName()));
+        return partitionNames;
+    }
+
+    @Override
+    public void analyze(List<ColumnDef> columnDefs, Map<String, String> tableProperties) throws AnalysisException {
+        // analyze partition columns
+        this.analyzePartitionColumns(columnDefs);
+        // analyze single list property
+        this.analyzeSingleListPartition(tableProperties);
+        // analyze multi list partition
+        this.analyzeMultiListPartition(tableProperties);
+    }
+
+    private void analyzePartitionColumns(List<ColumnDef> columnDefs) throws AnalysisException {
+        if (partitionColNames == null || partitionColNames.isEmpty()) {
+            throw new AnalysisException("No partition columns.");
+        }
+        Set<String> partColNames = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+        for (String partitionCol : partitionColNames) {
+            if (!partColNames.add(partitionCol)) {
+                throw new AnalysisException("Duplicated partition column " + partitionCol);
+            }
+            boolean found = false;
+            for (ColumnDef columnDef : columnDefs) {
+                if (columnDef.getName().equals(partitionCol)) {
+                    if (!columnDef.isKey() && columnDef.getAggregateType() != AggregateType.NONE) {
+                        throw new AnalysisException("The partition column could not be aggregated column");
+                    }
+                    if (columnDef.getType().isFloatingPointType() || columnDef.getType().isComplexType()) {
+                        throw new AnalysisException(String.format("Invalid partition column '%s': %s",
+                                columnDef.getName(), "invalid data type " + columnDef.getType()));
+                    }
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                throw new AnalysisException("Partition column[" + partitionCol + "] does not exist in column list.");
+            }
+        }
+    }
+
+    private void analyzeMultiListPartition(Map<String, String> tableProperties) throws AnalysisException {
+        Set<String> multiListPartitionName = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+        for (MultiListPartitionDesc desc : this.multiListPartitionDescs) {
+            if (!multiListPartitionName.add(desc.getPartitionName())) {
+                throw new AnalysisException("Duplicated partition name: " + desc.getPartitionName());
+            }
+            desc.analyze(partitionColNames.size(), tableProperties);
+        }
```

Comment:
It is also necessary to check whether there are duplicate values between [Multi|Single]ListPartitionDescs.